### PR TITLE
Fixed even more signedness and conversion issues

### PR DESCRIPTION
**Pull request by AtariDreams**
This should be the last of them.

### CODE REVIEW

File: lib/util/sudo_debug.c [hunk 1]
```diff
@@ -536,7 +536,7 @@ sudo_debug_exit_id_t_v1(const char *func, const char *file, int line,
 	"<- %s @ %s:%d := %lld", func, file, line, (long long)ret);
 #else
     sudo_debug_printf2(NULL, NULL, 0, subsys | SUDO_DEBUG_TRACE,
-	"<- %s @ %s:%d := %d", func, file, line, (int)ret);
+	"<- %s @ %s:%d := %ld", func, file, line, (long)ret);
```
Comment (millert) [inline lib/util/sudo_debug.c:539]: I don't see a need for this change.  If `id_t` is 32-bits, printing it as an int should be fine.

File: lib/util/sudo_debug.c [hunk 2]
```diff
@@ -565,7 +565,7 @@ sudo_debug_exit_time_t_v1(const char *func, const char *file, int line,
...
```
Comment (millert) [inline lib/util/sudo_debug.c:568]: Same here, a 32-bit time_t will fit into an int, there is no need to use a long.

File: plugins/sudoers/toke.c [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 #include <config.h>
 
-#line 3 "toke.c"
```
Comment (millert) [inline plugins/sudoers/toke.c:8]: Please drop changes to this file, your version of flex and mine appear to disagree on line numbers.

**Review by millert — CHANGES_REQUESTED** [review]

**Comment by AtariDreams** [comment]
@millert Done!

File: lib/iolog/iolog_mkdirs.c [hunk 4]
```diff
@@ -116,11 +116,12 @@ iolog_mkdirs(const char *path)
 	    if (errno == EACCES && !uid_changed) {
 		/* Try again as the I/O log owner (for NFS). */
 		uid_changed = iolog_swapids(false);
-		if (uid_changed)
+		if (uid_changed) {
```
Comment (millert) [inline lib/iolog/iolog_mkdirs.c:119]: The changes to this file should be dropped, the code is correct as-is.  We want to display a warning when the initial mkdirat() fails if we don't retry after swapping ids.

**Review by millert — CHANGES_REQUESTED** [review]

**Comment by AtariDreams** [comment]
@millert Done!

**Review by millert — APPROVED** [review]

**Pull request merged**